### DISCUSSION TITLE
Certificate for msi

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#65] [General]
Is there a chance to sign the `msi` installer so that I don't have to ask the Administrator for approval with each release? :)

**Comment by CEXT-Dan** [comment]
It’s something to consider, many companies require it. However, there are some issues.
I don’t want to pay money.
I really don’t know how it works.

The installer just copies files. I’d be willing to provide a ZIP file with the ARX binaries, you could just swap them out. 
Everything else is on GitHub. At least that would reduce the frequency of actually having to run the installer

**Comment by CEXT-Dan** [comment]
I could not find a free solution, so, I have no plans to do this just yet

All the installer does is create the folders and place files. it’s really just for convenience. 
Starting with the next release, I will include a zip file with the binaries.  I'll include a note if there are major changes 
Everything else is available here in the repository, hopefully this will help for now

**Reply by kwollaston** [reply]
Hi Dan, I am testing deployment on a machine I do not have admin access to and since I am able to install Python without admin permissions, I have written a install script in Python that will install win32com/wx and copy my pyrx_onload file.

I have been able to get PyRx installed and working without the installed based on the binaries you provided although a few extra steps were needed as I'm sure you know.

* I needed the PyRx.ini file (I don't know if it automatically detected Python or not, I filled it in myself)
* I had to separately download the Stubs
* I had to separately download the PackageContents.xml

Once I had these things in place, everything loaded up as expected automatically at startup.

Is there a way you can package a "portable" zip which has everything necessary?

**Reply by gswifort** [reply]
This is in the plans. Check #86 and #87.

**Reply by CEXT-Dan** [reply]
In the meantime, I’ve attempted to remove the need for admin rights
https://stackoverflow.com/questions/4080131/how-to-make-a-setup-work-for-limited-non-admin-users

I don't have a good way to test it though

**Reply by CEXT-Dan** [reply]
Can someone test if admin is still required with the v1.3.029 MSI installer?

**Reply by gswifort** [reply]
It works, no admin privileges are needed.

**Reply by CEXT-Dan** [reply]
Great, Thank you!
